### ISSUE TITLE
Share from gallery - problem with cropping

### ISSUE THREAD

**Issue by skandaI** [rodit/SnapMod#90]
I've noticed that when I send a snap from the gallery to my other phone (with different screen ratio) as a "raw file" everything is fine, but when I use editor for example to add some captions, when I open the snap on my other phone it's kind of cropped wrong and has black bars on the sides, which makes it look suspicious (like fake). Is there any way to prevent this? For now my best solution is to send snaps that already have captions rendered into the video. It looks good on my both phones when video has ratio 9:16.

**Comment by Magna-Graecia** [comment]
This isn't an issue with SnapMod. This happens normally through Snapchat regardless. Go to SnapMap and search for Paris, France, then view the many snaps uploaded from there. You will see tons of photos and videos that have black bars on the sides, and many that don't. I just did it right now with a 16:9 device. It's just different screen sizes. I know what you mean in saying that it looks suspicious, but this is because people are dumb and think all phones still have 16:9 ratio like they used to so snaps were always full screen. Now that's not true in 2022.

You can download Easy DPI Changer from the play store and use it to change your screen settings and see how it changes the way photos/videos are sent, but keep in mind that you're only seeing it on your devices with your specific screen ratios. You can't see how it looks on every single device out there, but I guess you can try to see how it looks on as many as you can by asking friends.

**Comment by skandaI** [comment]
@Magna-Graecia
Thank you for this wise comment.
However, I'm still wondering why those black bars on the sides don't show up on both phones when I send a snap without editing. In this case the bar only appears at the bottom and is not so conspicuous.
Phone 1: 20:9 ratio (2400x1080px)
Phone 2: 16:9 (1920x1080px)
Video ratio: 16:9
I think the Snapchat editor is doing something weird.

**Comment by Magna-Graecia** [comment]
No problem. I know what you mean, it does the same for me. I sent a photo without any edits, and then the same photo with a 3-letter caption. Without edits the photo showed as full screen, and with edits it had the black bars on the side. What I noticed was that without edits and at full screen, the photo was zoomed in. Snapchat basically cropped a bit of it out. With edits it showed the full photo, not cropped at all, and with black bars on the sides.

I believe it's because the photo was taken on a device with a longer screen than mine, so when editing with a caption, it needs to show the full photo because the gray Snapchat caption bar runs horizontally across the entire photo, and in order for it to show the full photo, it shrinks it a bit so there ends up being empty space on the sides. What you can do is try to play with Easy DPI Changer to put your device's screen in the same ratio as the screen of the device which the photo you're trying to send was taken on. I did this for photos/videos that were 16:9 because my device has a 1920:1200 ratio, and it would cause black bars to show. So I changed it to be 1920:1080 and it made the black bars go away. Trust me, play around with it and you will see.

**Comment by Magna-Graecia** [comment]
Just as an example, I just changed my screen ratio back to the default (1920x1200) and sent the same photo as before with a 3-letter caption. Not only did it show the black bars on the sides even bigger, but there was now a black bar on the bottom as you described. When I changed it back to 1920x1080 and sent with the same edit, it would no longer have the black bar on the bottom, and the bars on the sides were thinner. Try to match your screen ratio to the same as the screen of the device that took the photo/video you're trying to send, and you will see that it will be full screen.

In either case, this happens naturally with Snapchat as I explained if you view snaps on SnapMap, so it doesn't mean the snaps are fake, but I understand your concern. I also like when they're full screen as they look less funny.

**Comment by Magna-Graecia** [comment]
Yep, my theory is correct. I just tested it. I took that photo that gave me the black bars, and I knew it came from an iPhone 13 with a screen resolution of 2532x1170 pixels. I used Easy DPI Changer to imitate this ratio on my device. My device is only 1920x1200 (smaller), so what I did was I divided 1170 by 2532 to find out what percentage the width was of the height on iPhone 13. I then used this percentage to multiply by 1920 to find out what the width should be on my device in order to imimate the same ratio. The number was approximately 887. I used Easy DPI Changer to change my screen resolution to 1920x887 (it made the screen so much thinner lol, but this is how it imitates the elongated iPhone 13 screen). I then sent the same photo as before both without edits and with edits. They were full screen both times. Try it for yourself and you will see.

**Comment by skandaI** [comment]
Yes, I tried Easy DPI Changer and I have similar experiences. Maybe this is some idea for a new feature in SnapMod: the ability to change the aspect ratio of a file in the editor without changing the resolution of the whole system?

**Comment by Magna-Graecia** [comment]
You can't change the aspect ratio of the file without cropping it or distorting it. It was created with a certain pixel ratio, so you would need to either crop it to change that ratio, or stretch/shrink it, which will cause distortion.

Take the photo you want to send, find out it's pixel ratio by opening it in a photo editor that's as basic as Windows Paint, and crop the photo to match the pixel ratio of your device screen. You will see that it will now get sent as full screen, but of course some of the image has been cropped out. You can do the same with videos using a program called Handbrake. It allows you to crop the video from all 4 sides as you please.

I guess a feature could be the ability to mimic a custom screen resolution from the camera editor in Snapchat so you don't need to change it for the entire system, but this may be way too complicated to implement. I'm not sure. Rodit would have to comment on that.

**Comment by skandaI** [comment]
"when editing with a caption, it needs to show the full photo because the gray Snapchat caption bar runs horizontally across the entire photo, and in order for it to show the full photo"

Maybe the simplest solution would be to modify by SnapMod this text box so it doesn't require the file to be displayed in its full form? I mean I don't mind the photo/video being zoomed in. It would be great if it adjust itself and ignore the width of the gray bar.

**Comment by Magna-Graecia** [comment]
It happens with filters and perhaps other edits as well, so a lot of things would need to be modified for something like that. Personally I don't have much issue with the way it is now. If I really need to make sure a photo/video is going to display full screen, I'll just adjust my screen resolution to match the photo/video for that moment. It only takes 5 seconds to do it. I like the option of choosing how I want photos/videos to be sent as I have now, rather than to have SnapMod messing with the fit of the photo to make sure it goes full screen.

**Comment by rodit** [comment]
Sorry, this probably isn't something I plan on working on. If you show me what looks wrong on the recipient end (and the original photo/video being sent) I might be able to suggest a fix, however.
